sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DateExpressionsSuite.scala

```diff
@@ -122,8 +124,8 @@ class DateExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     (2000 to 2002).foreach { y =>
       (0 to 11 by 11).foreach { m =>
         c.set(y, m, 28)
-        (0 to 5 * 24).foreach { i =>
-          c.add(Calendar.HOUR_OF_DAY, 1)
+        (0 to 12).foreach { i =>
```

Comment:
Where is `i` used?

Reply:
It is not used.

Reply:
Why? What is the reason to test the same `HOUR_OF_DAY ` multiple times?

Reply:
> Why? What is the reason to test the same HOUR_OF_DAY multiple times?

The HOUR_OF_DAY is increasing in the loop. I am simply following the original logic.

Reply:
ah, I see. Thanks.